frontend/model/state.js

```diff
@@ -90,7 +124,13 @@ sbp('sbp/selectors/register', {
       // $FlowFixMe[incompatible-call]
       Vue.set(state, 'reverseNamespaceLookups', Object.fromEntries(Object.entries(state.namespaceLookups).map(([k, v]: [string, string]) => [v, k])))
     }
-    (() => {
+    contractUpdate((contractIDHints: ?string[]) => {
+      const state = sbp('state/vuex/state')
+      if (Array.isArray(contractIDHints)) {
+        if (!contractIDHints.reduce((acc, contractID) => {
+          return (acc || state.contracts[contractID]?.type === 'gi.contracts/group')
+        }, false)) return
+      }
```

Comment:
Seeing a lot of DRY violations with this chunk of code. Seems ugly too.

Is there a way that, instead of this blob, we can pass in some parameter to `contractUpdate` when calling it that results in the same sort of logic being run to fix the DRY violation and remove this block from here?